File: Subset Sums - GFG/subset-sums.cpp
//{ Driver Code Starts
#include<bits/stdc++.h> 
using namespace std; 

// } Driver Code Ends
class Solution
{
public:
    void ss(int ind,vector<int> &arr,vector<int> &res,int sum){
        if(ind == arr.size()){
            res.push_back(sum);
            return;
        }
        sum += arr[ind];
        ss(ind + 1,arr,res,sum);
        sum -= arr[ind];
        ss(ind + 1,arr,res,sum);
    }

    vector<int> subsetSums(vector<int> arr, int N)
    {
        // Write Your Code here
        vector<int> res;
        ss(0,arr,res,0);
        return res;
    }
};

//{ Driver Code Starts.
int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int N;
        cin>>N;
        vector<int> arr(N);
        for(int i = 0 ; i < N ; i++){
            cin >> arr[i];
        }
        Solution ob;
        vector<int> ans = ob.subsetSums(arr,N);
        sort(ans.begin(),ans.end());
        for(auto sum : ans){
            cout<< sum<<" ";
        }
        cout<<endl;
    }
    return 0;
}
// } Driver Code Ends

File: Subset Sums - GFG/subset-sums.py
#User function Template for python3
class Solution:
    def helper(self,ind,arr,sum,res):
        if ind == len(arr):
            res.append(sum)
            return
        self.helper(ind + 1,arr,sum + arr[ind],res)
        self.helper(ind + 1,arr,sum ,res)
    
	def subsetSums(self, arr, N):
		res = []
		self.helper(0,arr,0,res)
		return res


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    T=int(input())
    for i in range(T):
        N = int(input())
        arr = [int(x) for x in input().split()]
        ob = Solution()
        ans = ob.subsetSums(arr, N)
        ans.sort()
        for x in ans:
            print(x,end=" ")
        print("")

# } Driver Code Ends